src/Microcosm.js

```diff
@@ -20,104 +19,50 @@ let Microcosm = function() {
    */
   Diode(this)
 
-  /**
-   * Represents all "merged" transactions. Whenever a transaction completes,
-   * the result is folded into base state and the transaction object is
-   * "released". This lets transactions execute in a predicable order while
-   * not soaking up memory keeping them forever.
-   */
-  this.base = {}
-
-  /**
-   * Holds publically available state. The result of folding all incomplete
-   * transactions over base state. This property can safely be referenced when
-   * retrieving application state.
-   */
-  this.state = this.base
-
+  this.state   = {}
   this.stores  = []
   this.plugins = []
-  this.history = new Tree()
 
+  // Standard store reduction behaviors
   this.addStore(MetaStore)
+
+  // A place to keep all developer warnings
+  this.addPlugin(Debug)
+
+  // Manage key lifecycle hooks for state
+  this.addPlugin(State)
+
+  // Keep track of history as a tree
+  this.addPlugin(History, options)
 }
 
 Microcosm.prototype = {
   constructor: Microcosm,
 
   getInitialState() {
-    return dispatch(this.stores, {}, Transaction(lifecycle.willStart, this.state))
+    return this.lifecycle(lifecycle.willStart)
   },
 
-  /**
-   * Dispatch all open transactions upon base state to determine
-   * a new state. This is the state exposed to the outside world.
-   */
   rollforward() {
-    this.state = this.history.branch().reduce((state, transaction) => {
-      return dispatch(this.stores, state, transaction)
-    }, this.base)
+    this.state = this.lifecycle(lifecycle.willRollforward)
 
     this.emit(this.state)
 
     return this
   },
 
-  shouldHistoryKeep(transaction) {
-    return false
-  },
-
-  clean(transaction) {
-    if (transaction.complete && !this.shouldHistoryKeep(transaction)) {
-      this.base = dispatch(this.stores, this.base, transaction)
-      return true
-    }
-    return false
-  },
-
-  transactionWillOpen(transaction) {
-    this.history.append(transaction)
-  },
-
-  transactionWillUpdate(transaction, error, payload, complete) {
-    transaction.active   = !error
-    transaction.complete = complete
-    transaction.error    = error
-    transaction.payload  = payload
-  },
-
-  transactionWillClose() {
-    this.history.prune(this.clean, this)
+  lifecycle(type, initial={}) {
```

Comment:
thinking `stage`, or `eventType`, or `hook` might be a more descriptive name for `type`